src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -775,37 +790,124 @@ static BOOL AttributeHasChangesOmittedQuality(MTRAttributePath * attributePath)
     // 4. Cache has no entry
     // TODO: add option for BaseSubscriptionCallback to report during priming, to reduce when case 4 is hit
     if (!attributeIsSpecified || ![self _subscriptionAbleToReport] || hasChangesOmittedQuality || !attributeValueToReturn) {
+        // Read requests container will be a mutable array of items, each being an array containing:
+        //   [endpoint ID, cluster ID, attribute ID, params]
```

Comment:
Is there a reason to use an array with predefined indices instead of just defining an interface type for the entries?  But I guess fine for now.

Reply:
Was just because it's simpler and really just used in one spot. I think this could use a proper interface if the code around here gets messier / harder to keep track of things.